version: '3.8'

services:
#  main:
#    container_name: main
#    build:
#      context: .
#    env_file:
#      - .dev.env
#    volumes:
#      - .:/app
#      - /app/node_modules
#    ports:
#      - 5000:5000
#      - 9229:9229
#    command: yarn start:dev
#    depends_on:
#      - postgres
#    restart: always
#    networks:
#      - postgres

  postgres_db:
    container_name: postgres_db
    image: postgres:14.0-alpine
#    env_file:
#      - .dev.env
    environment:
      - POSTGRES_PASSWORD=${TYPEORM_PASSWORD}
      - POSTGRES_USER=${TYPEORM_USERNAME}
      - POSTGRES_DB=${TYPEORM_DATABASE}
      - POSTGRES_HOST=${DB_HOST}
    ports:
      - ${TYPEORM_PORT}:5432
    volumes:
      - ./pgdata:/var/lib/postgresql/data
    restart: on-failure
    networks:
      - db_env

  pgadmin:
      container_name: pgadmin_container-po
      image: dpage/pgadmin4:6.12
      logging:
        driver: none
      environment:
        PGADMIN_DEFAULT_EMAIL: "askal@ukr.net"
        PGADMIN_DEFAULT_PASSWORD: "postgre"
        PGADMIN_CONFIG_SERVER_MODE: "False"
      volumes:
        - ./pgadmin:/var/lib/pgadmin
      ports:
        - "5050:80"
      restart: on-failure
      depends_on:
        - postgres_db
      networks:
        - db_env
networks:
  db_env:
    driver: bridge



